src/pudl/analysis/demand_mapping.py

```diff
@@ -1,294 +1,561 @@
+            NAICS_DESC=lambda x: x.NAICS_DESC.astype(pd.StringDtype()),
+            SOURCE=lambda x: x.SOURCE.astype(pd.StringDtype()),
+            VAL_METHOD=lambda x: x.VAL_METHOD.astype(pd.StringDtype()),
+            WEBSITE=lambda x: x.WEBSITE.astype(pd.StringDtype()),
+            ABBRV=lambda x: x.ABBRV.astype(pd.StringDtype()),
+            YEAR=lambda x: x.YEAR.astype(pd.Int64Dtype()),
+            PEAK_LOAD=lambda x: x.PEAK_LOAD.astype(float),
+            PEAK_RANGE=lambda x: x.PEAK_RANGE.astype(float),
+            SHAPE_Length=lambda x: x.SHAPE_Length.astype(float),
+            SHAPE_Area=lambda x: x.SHAPE_Area.astype(float),
+        )
+    )
+    return gdf
 
-    new_df = (new_df[[gdf_intermediate_col, gdf_source_col, "area_derived", "geometry"]]
-              .merge(gdf_intermediate[[gdf_intermediate_col, "intermediate_area_derived"]])
-              .merge(gdf_source[[gdf_source_col, "source_area_derived"]]))
 
-    new_df["gdf_intermediate_intersection_fraction"] = new_df["area_derived"] / \
-        new_df["intermediate_area_derived"]
-    new_df["gdf_source_intersection_fraction"] = new_df["area_derived"] / \
-        new_df["source_area_derived"]
+################################################################################
+# Demand allocation functions
+################################################################################
+def edit_id_set(row, new_ID, ID):
+    """
+    Editing ID sets by adding the new geometry ID if required.
 
-    # Deleting temporary columns defined in the original GeoDataFrames
-    del gdf_intermediate["intermediate_area_derived"]
-    del gdf_source["source_area_derived"]
+    This function edits original ID sets by adding the new geometry ID if
+    required. This function is called by another function
+    `complete_disjoint_geoms`
+    """
+    if row["geom_type"] == "geometry_new_int":
+        return frozenset(list(row[ID]) + [new_ID])
 
-    new_df = new_df[[gdf_intermediate_col, gdf_source_col,
-                     "gdf_intermediate_intersection_fraction",
-                     "gdf_source_intersection_fraction"]]
+    else:
+        return row[ID]
 
-    return new_df
 
+def polygonize_geom(geom):
+    """
+    Remove zero-area geometries from a geometry collection.
 
-def create_intersection_matrix(gdf_intersection,
-                               gdf_intersection_col="gdf_intermediate_intersection_fraction",
-                               gdf_intermediate_col="FIPS",
-                               gdf_source_col="ID",
-                               normalization=True,
-                               normalize_axis=1):
+    Strip zero-area geometries from a geometrical object.
+    (maybe a single geometry object or collection e.g. GeometryCollection)
+    This function is called by another function `complete_disjoint_geoms`.
     """
-    Pivots stacked dataframe to an intersection matrix.
+    if type(geom) == GeometryCollection:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5554119)